source/core/shape.cc

```diff
@@ -481,20 +501,222 @@ CompositeShape<dim>::static_copy() const
+      << std::endl;
+  return bounding_box->displaced_volume(fluid_density);
+}
+
+template <int dim>
+double
+RBFShape<dim>::wendlandc2(const double distance) const
+{
+  return distance > 1.0 ? 0.0 :
+                          std::pow(1. - distance, 4.0) * (4.0 * distance + 1.0);
+}
+
+template <int dim>
+double
+RBFShape<dim>::linear(const double distance) const
+{
+  return distance > 1.0 ? 0.0 : (1.0 - distance);
+}
+
+template <int dim>
+double
+RBFShape<dim>::gauss90(const double distance) const
+{
+  double eps = std::pow(-1.0 * std::log(0.1), 0.5);
+  return std::exp(-1.0 * std::pow(distance * eps, 2.0));
+}
+
+template <int dim>
+double
+RBFShape<dim>::gauss95(const double distance) const
+{
+  double eps = std::pow(-1.0 * std::log(0.05), 0.5);
+  return std::exp(-1.0 * std::pow(distance * eps, 2.0));
+}
+
+template <int dim>
+double
+RBFShape<dim>::gauss99(const double distance) const
+{
+  double eps = std::pow(-1.0 * std::log(0.01), 0.5);
+  return std::exp(-1.0 * std::pow(distance * eps, 2.0));
+}
+
+template <int dim>
+double
+RBFShape<dim>::c1c0(const double distance) const
+{
+  return distance > 1.0 ? 0.0 : (1.0 - std::pow(distance, 2.0));
+}
+
+template <int dim>
+double
+RBFShape<dim>::c2c0(const double distance) const
+{
+  return distance > 1.0 ? 0.0 : (1.0 - std::pow(distance, 3.0));
+}
+
+template <int dim>
+double
+RBFShape<dim>::c0c1(const double distance) const
+{
+  return distance > 1.0 ? 0.0 :
+                          (1.0 - 2.0 * distance + std::pow(distance, 2.0));
+}
+
+template <int dim>
+double
+RBFShape<dim>::c1c1(const double distance) const
+{
+  return distance > 1.0 ? 0.0 :
+                          (1.0 - 3.0 * std::pow(distance, 2.0) +
+                           2.0 * std::pow(distance, 3.0));
+}
+
+template <int dim>
+double
+RBFShape<dim>::c2c1(const double distance) const
+{
+  return distance > 1.0 ? 0.0 :
+                          (1.0 - 4.0 * std::pow(distance, 3.0) +
+                           3.0 * std::pow(distance, 4.0));
+}
+
+template <int dim>
+double
+RBFShape<dim>::c0c2(const double distance) const
+{
+  return distance > 1.0 ?
+           0.0 :
+           (1.0 - 3.0 * distance + 3.0 * std::pow(distance, 2.0) -
+            std::pow(distance, 3.0));
+}
+
+template <int dim>
+double
+RBFShape<dim>::c1c2(const double distance) const
+{
+  return distance > 1.0 ?
+           0.0 :
+           (1.0 - 6.0 * std::pow(distance, 2.0) +
+            8.0 * std::pow(distance, 3.0) - 3.0 * std::pow(distance, 4.0));
+}
+
+template <int dim>
+double
+RBFShape<dim>::c2c2(const double distance) const
+{
+  return distance > 1.0 ?
+           0.0 :
+           (1.0 - 10.0 * std::pow(distance, 3.0) +
+            15.0 * std::pow(distance, 4.0) - 6.0 * std::pow(distance, 5.0));
+}
+
+template <int dim>
+double
+RBFShape<dim>::evaluate_basis_function(const RBFBasisFunction basis_function,
+                                       const double           distance) const
```

Comment:
Would it have been possible instead of using a switch, to template the kernel type in the shape ? 
Does not need to be done now, but keep this in mind.

Reply:
I'll keep this in mind, but first I'll check if it would be useful to have different kernel types for different nodes. That is probably not useful though, just an idea.